Sources/GraphQLWebSocket/WebSocketClient.swift

```diff
@@ -0,0 +1,200 @@
+        assert(request.allHTTPHeaderFields?[HTTPWSHeader.protocolName] != nil)
+        socket = session.webSocketTask(with: request)
+        self.callbackQueue = callbackQueue
+    }
+
+    /// Starts receiving incoming messages.
+    private func receive() {
+        socket.receive { [weak self] result in
+            guard let self else { return }
+            self.callbackQueue.async {
+                switch result {
+                case let .success(message):
+                    switch message {
+                    case let .data(data):
+                        self.delegate?.didReceive(event: .binary(data), client: self)
+                    case let .string(string):
+                        self.delegate?.didReceive(event: .text(string), client: self)
+                    @unknown default:
+                        assertionFailure("Unknown message type")
+                    }
+                case let .failure(error):
+                    self.delegate?.didReceive(event: .error(error), client: self)
+                }
+            }
+            self.receive()
+        }
+    }
+    
+    // MARK: WebSocketClient
+    
+    func connect() {
+        assert(socket.state == .suspended)
+        socket.delegate = self
+        receive()
+        socket.resume()
+    }
+    
+    func disconnect(closeCode: UInt16) {
+        assert(socket.state == .running)
+        // NOTE: URLSessionWebSocketTask.CloseCode limits allowed close codes
+        let closeCode = URLSessionWebSocketTask.CloseCode(rawValue: Int(closeCode)) ?? .normalClosure
+        socket.cancel(with: closeCode, reason: nil)
+    }
+    
+    func write(string: String, completion: (() -> ())?) {
+        assert(socket.state == .running)
+        socket.send(.string(string)) { error in
+            // NOTE: Starscream NativeEngine ignores the error
+            assert(error == nil)
+            completion?()
+        }
+    }
+
+    func write(data: Data, completion: (() -> ())?) {
+        assert(socket.state == .running)
+        socket.send(.data(data)) { error in
+            // NOTE: Starscream NativeEngine ignores the error
+            assert(error == nil)
+            completion?()
+        }
+    }
+    
+    func write(ping: Data, completion: (() -> ())?) {
+        assert(socket.state == .running)
+        socket.sendPing(pongReceiveHandler: { error in
+            // NOTE: Starscream NativeEngine ignores the error
+            assert(error == nil)
+            completion?()
+            guard error == nil else { return }
+            self.callbackQueue.async { [weak self] in
+                guard let self else { return }
+                self.delegate?.didReceive(event: .pong(nil), client: self)
+            }
+        })
+    }
+    
+    func write(pong: Data, completion: (() -> ())?) {
+        assert(socket.state == .running)
+        // TODO: test, check if URLSessionWebSocketTask responds to ping
```

Comment:
I really appreciate you opening a PR to replace Starscream, but I'm confused why there's a TODO left here?